fs/metrics.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+#include "rocksdb/env.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+struct ZenFSMetrics {
+public:
+  ZenFSMetrics() {}
+  virtual ~ZenFSMetrics() {}
+public:
+  // Add a reporter named label.
+  // You can give a type for type-checking.
+  virtual void AddReporter(uint32_t label, uint32_t type = 0) = 0;
```

Comment:
What does type-checking mean?

Reply:
Different Reporter may have different types, and even if the values are of the same type, the units may be different. For example, we may have created a Reporter with QPS but called it with throughput X passed in. to prevent errors caused by this situation, we plan not to use Report() directly, but use ReportXXXX() with type-checking instead.

Reply:
Should the `type` be a enum? e.g. `TYPE_TIME_MS`, `TYPE_QPS`, etc.

Reply:
Yes, it should.

Reply:
Just found `ZenFSMetricsReporterType`. Better to document it in the comments.